Lazy Loading Config

mileslucas opened the pull request:
moved config.yaml to base directory, added config.py which uses a lazy-loading scheme for config parsing and replaced all appropriate occurrences of Starfish.<group> with Starfish.config.<group>.

Issue #104

mileslucas commented:
I've piggy-backed a couple other changes onto this with regards to the installation. I've rewritten `setup.py` to be less dense and a little more intelligent. You no longer have to install numpy or cython before installing this package using a cool build_ext trick. It is also completely installable from pip either as a cloned repo or straight from github (#95 ). I've also updated the README to reflect these changes. This should be ready to go, but feel free to poke around.

iancze commented on Starfish/config.py:
```diff
@@ -0,0 +1,40 @@
+            return super().__getattribute__(item)
+        with open(self.filename) as f:
+            base = yaml.safe_load(f)
+            if item == 'name':
```
`namen` -> `name`?

mileslucas replied:
Yes 👍

iancze commented:
This is some great functionality to the config files, thanks! Also the cython install fix sounds like it will really help smooth things out for people. Once I make the `develop` branch let's merge this there.

mileslucas commented:
Added a unit test under `tests/test_config.py` and found some more bugs. Yay test driven development! I've fixed the bugs and confirmed the tests pass. It works in isolation, like
```shell
$ pytest tests/test_config.py
```
